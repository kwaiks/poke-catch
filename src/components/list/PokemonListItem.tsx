import { listItemStyle } from "styles/emotion/list";
import { Pokemon } from "types/pokemon";
import Image from "next/image";
import Link from "next/link";
import { usePokemon } from "context/pokemonContext";

interface Props {
    pokemon: Pokemon;
    extra: string;
    children?: React.ReactChild;
}

export default function PokemonListItem({pokemon, extra, children}:Props){
    const {setSelectedPokemon} = usePokemon();

    const viewPokemon = () => {
        setSelectedPokemon(pokemon);
    }

    return (
        <Link href={`/pokemon/${pokemon.name}`} passHref>
            <div 
            data-testid="pokemon-list-item"
            onClick={viewPokemon}
            className={listItemStyle}>
                <div>
                    <Image 
                    data-testid="pokemon-list-item-image"
                    src={pokemon.image} height="80px" width="80px" alt={pokemon.name}/>
                </div>
                <div style={{width: "100%"}}>
                    <div 
                    data-testid="pokemon-list-item-name"
                    style={{
                        textTransform: "capitalize",
                        fontWeight: "bold"}}>
                    {pokemon.name}
                    </div>
                    <div
                    data-testid="pokemon-list-item-extra"
                    >
                        {extra}
                    </div>
                    <div
                    data-testid="pokemon-list-item-children"
                    >
                        {children}
                    </div>
                </div>
            </div>
        </Link>
    )
}