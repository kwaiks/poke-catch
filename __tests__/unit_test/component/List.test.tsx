import '@testing-library/jest-dom';
import { create } from "react-test-renderer";
import { render, screen } from '@testing-library/react';
import PokemonMovesWrapper from 'components/list/MovesWrapper';
import PokemonListItem from 'components/list/PokemonListItem';
import ShimmerList from 'components/list/ShimmerListItem';


describe(("List Test"), () => {

    describe("Moves Wrapper", () => {
        it("should render",()=> {
            render(<PokemonMovesWrapper moves={[]}/>)
            expect(screen.getByTestId("list-pokemon-wrapper")).toBeVisible()
            expect(screen.getByTestId("list-pokemon-wrapper-title")).toBeVisible()
            expect(screen.getByTestId("list-pokemon-wrapper-list")).toBeVisible()
        })

        it("should match snapshot", () => {
            const wrapper = create(<PokemonMovesWrapper moves={[]}/>).toJSON();

            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("Pokemon List Item", () => {
        const pokemon = {
            name: "Bulbasaur",
            image: "/img",
        }

        const extra = "Owned"

        it("should render and match params", () => {
            render(<PokemonListItem
                pokemon={pokemon}
                extra={extra}
            />);
            expect(screen.getByTestId("pokemon-list-item")).toBeVisible();
            expect(screen.getByTestId("pokemon-list-item-image")).toBeVisible();

            const title = screen.getByTestId("pokemon-list-item-name");
            expect(title).toBeVisible();
            expect(title.textContent).toEqual(pokemon.name);
            
            const extraTest = screen.getByTestId("pokemon-list-item-extra");
            expect(extraTest).toBeVisible();
            expect(extraTest.textContent).toEqual(extra); 
        })

        it("should match snapshot", () => {
            const wrapper = create(<PokemonListItem
                pokemon={pokemon}
                extra={extra}
            />).toJSON();

            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("Shimmer List Item", () => {
        it("should render", () => {
            render(<ShimmerList/>)

            const wrapper = screen.getByTestId("shimmer-list-wrapper");
            expect(wrapper).toBeVisible();
            expect(wrapper.childElementCount).toEqual(8);

            expect(screen.getAllByTestId("shimmer-item")[0]).toBeVisible();
            expect(screen.getAllByTestId("shimmer-image")[0]).toBeVisible();
            expect(screen.getAllByTestId("shimmer-title")[0]).toBeVisible()
            expect(screen.getAllByTestId("shimmer-subtitle")[0]).toBeVisible()
        })

        it("should match snapshot", () => {
            const wrapper = create(<ShimmerList/>).toJSON();

            expect(wrapper).toMatchSnapshot()
        })
    })


})