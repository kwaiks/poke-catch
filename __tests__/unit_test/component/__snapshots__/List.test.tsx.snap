// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Test Moves Wrapper should match snapshot 1`] = `
<div
  data-testid="list-pokemon-wrapper"
>
  <div
    className="css-1cz2e7u"
    data-testid="list-pokemon-wrapper-title"
  >
    <span
      className="css-knzy0n"
    >
      Moves
    </span>
  </div>
  <div
    className="css-rwbjee"
    data-testid="list-pokemon-wrapper-list"
  />
</div>
`;

exports[`List Test Pokemon List Item should match snapshot 1`] = `
<div
  className="css-1px4tnp"
  data-testid="pokemon-list-item"
  href="/pokemon/Bulbasaur"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iODAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          alt="Bulbasaur"
          data-testid="pokemon-list-item-image"
          decoding="async"
          src="/_next/image?url=%2Fimg&w=256&q=75"
          srcSet="/_next/image?url=%2Fimg&w=96&q=75 1x, /_next/image?url=%2Fimg&w=256&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        alt="Bulbasaur"
        data-testid="pokemon-list-item-image"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      data-testid="pokemon-list-item-name"
      style={
        Object {
          "fontWeight": "bold",
          "textTransform": "capitalize",
        }
      }
    >
      Bulbasaur
    </div>
    <div
      data-testid="pokemon-list-item-extra"
    >
      Owned
    </div>
    <div
      data-testid="pokemon-list-item-children"
    />
  </div>
</div>
`;

exports[`List Test Shimmer List Item should match snapshot 1`] = `
<div
  className="css-1dgswtx"
  data-testid="shimmer-list-wrapper"
>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
  <div
    className="css-1px4tnp"
    data-testid="shimmer-item"
  >
    <div
      className="css-1681m75 css-1wdd5jf"
      data-testid="shimmer-image"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-title"
    />
    <div
      className="css-1681m75 css-1a1a8ro"
      data-testid="shimmer-subtitle"
    />
  </div>
</div>
`;
